base_layer/wallet/src/output_manager_service/storage/sqlite_db/mod.rs

```diff
@@ -415,64 +416,180 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
+    // exists, the specified columns (`mined_height`, `mined_in_block`, `status`, `mined_timestamp`,
+    // `marked_deleted_at_height`, `marked_deleted_in_block`, `last_validation_timestamp`) will be updated with the
+    // provided values. The `UPDATE` action updates the existing row with the new values provided by the
+    // `INSERT INTO` statement. The `excluded` keyword refers to the new data being inserted or updated and allows
+    // accessing the values provided in the `VALUES` clause of the `INSERT INTO` statement.
+    // Note:
+    //   `diesel` does not support batch updates, so we have to do it manually. For example, this
+    //   `diesel::insert_into(...).values(&...).on_conflict(outputs::hash).do_update().set((...)).execute(&mut conn)?;`
+    //   errors with
+    //   `the trait bound `BatchInsert<Vec<....>` is not satisfied`
+    fn set_received_outputs_mined_height_and_statuses(
         &self,
-        hash: FixedHash,
-        mined_height: u64,
-        mined_in_block: FixedHash,
-        confirmed: bool,
-        mined_timestamp: u64,
+        updates: Vec<ReceivedOutputInfoForBatch>,
     ) -> Result<(), OutputManagerStorageError> {
         let start = Instant::now();
         let mut conn = self.database_connection.get_pooled_connection()?;
         let acquire_lock = start.elapsed();
-        let status = if confirmed {
-            OutputStatus::Unspent as i32
-        } else {
-            OutputStatus::UnspentMinedUnconfirmed as i32
-        };
+
         debug!(
             target: LOG_TARGET,
-            "`set_received_output_mined_height` status: {}", status
+            "`set_received_outputs_mined_height_and_statuses` for {} outputs",
+            updates.len()
         );
-        let hash = hash.to_vec();
-        let mined_in_block = mined_in_block.to_vec();
-        let timestamp = NaiveDateTime::from_timestamp_opt(mined_timestamp as i64, 0).ok_or(
-            OutputManagerStorageError::ConversionError {
-                reason: format!("Could not create timestamp mined_timestamp: {}", mined_timestamp),
-            },
-        )?;
-        diesel::update(outputs::table.filter(outputs::hash.eq(hash)))
-            .set((
-                outputs::mined_height.eq(mined_height as i64),
-                outputs::mined_in_block.eq(mined_in_block),
-                outputs::status.eq(status),
-                outputs::mined_timestamp.eq(timestamp),
-                outputs::marked_deleted_at_height.eq::<Option<i64>>(None),
-                outputs::marked_deleted_in_block.eq::<Option<Vec<u8>>>(None),
-                outputs::last_validation_timestamp.eq::<Option<NaiveDateTime>>(None),
-            ))
-            .execute(&mut conn)
-            .num_rows_affected_or_not_found(1)?;
+
+        let mut query = String::from(
+            "INSERT INTO outputs ( commitment, mined_height, mined_in_block, status, mined_timestamp, spending_key, \
```

Comment:
One problem here is that we make a mistake here and give it something that does not exist, we will add it to the DB. 
And thus rather than fail on the update, its going to write junk into the database breaking the database. 

Something like a join union should work as well and not have this problem